### ISSUE TITLE
`no-redeclare` should not apply to a type and a value

### ISSUE THREAD

**Issue by ngolin** [eslint/eslint#13720]
**What rule do you want to change?**

`no-redeclare`.

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior.

**Please provide some example code that this change will affect:**

```ts
export declare namespace ControlBar {
  export interface Props {
    playing: boolean;
  }
}

// eslint-disable-next-line no-redeclare
export default function ControlBar({ playing }: ControlBar.Props) {
  return <div>Playing</div>;
}
```

**What does the rule currently do for this code?**

Raise `no-redeclare` error.

**What will the rule do after it's changed?**

Nothing.

**Are you willing to submit a pull request to implement this change?**

Not sure.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

Can you share your eslint config. As eslint doesnt support typescripts. You need to use the `typescript-eslint` as parser and use its [no-redeclare](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-redeclare.md) rule and turn off the base rule

**Comment by nzakas** [comment]
@anikethsaha don't forget to add the “evaluating” label to issues you’re looking at.

**Comment by anikethsaha** [comment]
yea sure
